src/network/network_chat_gui.cpp

```diff
@@ -261,20 +261,55 @@ static void SendChat(const std::string &buf, DestType type, int dest)
 		NetworkServerSendChat((NetworkAction)(NETWORK_ACTION_CHAT + type), type, dest, buf, CLIENT_ID_SERVER);
 	}
 }
+class NetworkChatAutoCompletion final : public AutoCompletion {
+public:
+	using AutoCompletion::AutoCompletion;
+
+private:
+	std::vector<std::string> GetSuggestions([[maybe_unused]] std::string_view prefix, std::string_view query) override
+	{
+		std::vector<std::string> suggestions;
+		for (NetworkClientInfo *ci : NetworkClientInfo::Iterate()) {
+			if (ci->client_name.starts_with(query)) {
+				suggestions.push_back(ci->client_name);
+			}
+		}
+		for (const Town *t : Town::Iterate()) {
+			/* Get the town-name via the string-system */
+			SetDParam(0, t->index);
+			std::string town_name = GetString(STR_TOWN_NAME);
+			if (town_name.starts_with(query)) {
+				suggestions.push_back(std::move(town_name));
+			}
+		}
+		return suggestions;
+	}
+
+	void ApplySuggestion(std::string_view prefix, std::string_view suggestion) override
+	{
+		if (prefix.empty()) {
+			this->textbuf->Assign(fmt::format("{}: ", suggestion));
+		} else {
+			this->textbuf->Assign(fmt::format("{}{} ", prefix, suggestion));
+		}
```

Comment:
similar

Reply:
So here it is actually important, as we add ": " if there is no prefix.
This is consistent with existing behaviour:  https://github.com/OpenTTD/OpenTTD/blob/626ee4ab4184068a9779813295a9dace14c3cc82/src/network/network_chat_gui.cpp#L416

Reply:
Maybe copy the old comment in this new location then.
And old code was `{} {}` for the other case.

Reply:
In new code prefix includes the separator space, so there is no need to add it in format string.
Added the comment.